maths/gamma_recursive.py

```diff
@@ -0,0 +1,82 @@
+"""
+Gamma function is a very useful tool in physics.
+It helps calculating complex integral in a convenient way.
+for more info: https://en.wikipedia.org/wiki/Gamma_function
+"""
+
+# Importing packages
+from math import sqrt, pi
+from re import match
+from typing import Union
+
+
+def gamma(num : Union[int, float]) -> Union[int, float]:
+    """
+    Calculates the value of Gamma function of num
+    where num is either an integer (1,2,3..) or a half-integer (0.5,1.5,2.5...).
+    Implemented using recursion
+    Examples:
+    >>> gamma(0.5)
+    1.7724538509055159
+    >>> gamma(2)
+    1
+    >>> gamma(3.5)
+    3.3233509704478426
```

Comment:
We need some doctest that raise ValueErrors so that we not only see how the algorithm work but also how it fails.

Reply:
Within the gamma function I didn't check the type of the argument. I did it in the main program before calling the function, so in the doctest I included negative numbers as arguments, but not strings. Let me know if that works or need to be modified.

Thanks